### PR TITLE
Migrate the ci (V3) to use KuzDoc

### PR DESCRIPTION
**Pull request by xbill82**
## What does this PR do?
Migrate the CI to use KuzDoc.

### How should this be manually tested?

Go take a look at the CI job for the dead-link check.

To test the build and deploy, you can follow the steps of the Doc Dev Deploy in the .travis.yml file and test them locally.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -72,12 +82,27 @@ jobs:
...
-        - gradle check
+        provider: script
+        script:
+          - npm run doc-upload
+        skip_cleanup: true
+
+      after_deploy:
+        - npm run doc-cloudfront
+
+    - stage: Deployment
```
Comment (Aschen) [inline .travis.yml]: Where does this stage come from?

Comment (xbill82) [reply .travis.yml]: IDK

Comment (Aschen) [reply .travis.yml]: @jenow Is it you?

Comment (jenow) [reply .travis.yml]: It has always been here https://github.com/kuzzleio/sdk-java/blob/master/.travis.yml#L97
Maybe see with @alexandrebouthinon?

Comment (Aschen) [reply .travis.yml]: I just thought that @xbill82 add it.  
Are we gonna deploy the v3 of the SDK with this now @alexandrebouthinon ?

**Comment by Aschen** [comment]
@xbill82 There is conflict on this PR

**Comment by Aschen** [comment]
@xbill82 can you fix the merge conflicts please?

**Pull request merged**